### PR TITLE
FilterBarSingleSelect

### PR DESCRIPTION
**Pull request by thomasdax98**
Implements a `FilterBarSingleSelect` based on the Mui Select component

https://deploy-preview-538--comet-admin.netlify.app/?path=/story/comet-admin-table-filterbar--filterbar-single-select

Closed:
Bildschirmfoto 2021-12-14 um 10 30 40

Open:
Bildschirmfoto 2021-12-14 um 10 31 04

Hover Item:
Bildschirmfoto 2021-12-14 um 10 31 08

### CODE REVIEW

**Comment by nsams** [comment]
This feature is much different as https://github.com/vivid-planet/comet-admin/pull/525

I think it should look and feel the same, except that it has a single select.

also we have to support async options (https://github.com/vivid-planet/comet-admin/pull/525 doesn't have that yet as well)

**Comment by thomasdax98** [comment]
@nsams I looked at https://github.com/vivid-planet/comet-admin/pull/525 and implemented it similarily

Async support is still missing

**Comment by nsams** [comment]
> @nsams I looked at #525 and implemented it similarily

You are not using FilterBarPopoverFilter.

We should make a call and talk about what we actually want and need!

**Comment by thomasdax98** [comment]
> You are not using FilterBarPopoverFilter.

Two reasons:

- It's not possible to hide the buttons inside FilterBarPopoverFilter (https://github.com/vivid-planet/comet-admin/blob/main/packages/admin/src/table/filterbar/filterBarPopoverFilter/FilterBarPopoverFilter.tsx#L81)
- It's not possible to close the FilterBarPopoverFilter from the outside. That would be needed for closing the FilterBarPopoverFilter after click on an item in the SingleSelect

> We should make a call and talk about what we actually want and need!

Yes!

File: packages/admin/src/table/filterbar/filterBarSingleSelect/FilterBarSingleSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                    </>
+                ),
+            });
+        });
+    }
+
+    return (
+        <div className={classes.wrapper}>
+            <Select
+                className={classes.root}
```
Comment (jamesricky) [inline packages/admin/src/table/filterbar/filterBarSingleSelect/FilterBarSingleSelect.tsx:45]: I think `root` should always be the outermost element... maybe like this:

```suggestion
        <div className={classes.root}>
            <Select
                className={classes.select}
```

File: packages/admin/src/table/filterbar/filterBarSingleSelect/FilterBarSingleSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                IconComponent={ChevronDown}
+                MenuProps={{
+                    MenuListProps: {
+                        className: classes.menuList,
+                    },
+                    className: classes.menu,
+                    PaperProps: { style: { marginTop: 2, marginLeft: -1 } },
```
Comment (jamesricky) [inline packages/admin/src/table/filterbar/filterBarSingleSelect/FilterBarSingleSelect.tsx:52]: Could this be done with a `className` ("paper" or "menuPaper") instead, to allow overriding the styles?

```suggestion
                    PaperProps: { className: classes.paper },
```

**Pull request closed without merging**